### DISCUSSION TITLE
Uniled Version 2 .1 (Beta) - Zengge (Hao Deng) BLE Mesh Support

### DISCUSSION THREAD

**Discussion by monty68** [monty68/uniled#43] [Announcements]
⭐ The next release of UniLED will add support for Zengge BLE mesh devices controlled by the Hao Deng app.

Initially, it will have limited device support as I only have a single CCT bulb, a few RGBW controllers, and a 4 group panel.

I'd be interested to hear from those people in the community who have any of these devices.

💥Breaking Changes

From this release 2.1, UniLED is unable to automatically migrate entities from previous versions and once installed, any entities previously created, will not function and must be deleted, rediscovered, and reinstalled.

**Comment by selfdestructer** [comment]
Glad I started following your repo when I found it looking for sp601e firmware. Literally just got a Zengge BLE device and I was not interested in downloading their app, at all. Looking forward to the new release! Thanks!

**Reply by monty68** [reply]
Unfortunately, I think  you still need the app to set up the mesh and UniLED still needs access to your cloud account to retrieve the mesh key, password and token to be able to connect. I'll run some tests to see if it will work without.

**Comment by sepqs** [comment]
Hi. Can you count on SP602E support? :)

**Reply by monty68** [reply]
Yes, currently working on it

**Reply by monty68** [reply]
Support now added into v2.2.0
